pkg/onboarding/usecases/metric/metrics_integration_test.go

```diff
@@ -0,0 +1,70 @@
+package metric_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/savannahghi/onboarding-service/pkg/onboarding/application/dto"
+	"github.com/savannahghi/onboarding-service/pkg/onboarding/domain"
+	"github.com/segmentio/ksuid"
+	"gorm.io/datatypes"
+)
+
+func TestUsecaseMetricsImpl_CollectMetrics_Integration(t *testing.T) {
+	ctx := context.Background()
+
+	m := testInfrastructureInteractor
+
+	metric := &dto.MetricInput{
+		Type:      domain.EngagementMetrics,
+		Payload:   datatypes.JSON([]byte(`{"who": "test user", "keyword": "suicidal"}`)),
+		Timestamp: time.Now(),
+		UID:       ksuid.New().String(),
+	}
+
+	invalidMetric := &dto.MetricInput{
+		Type:      "",
+		Payload:   datatypes.JSON([]byte(`{"who": "test user", "keyword": "bored"}`)),
+		Timestamp: time.Now(),
+		UID:       ksuid.New().String(),
+	}
+
+	type args struct {
+		ctx    context.Context
+		metric *dto.MetricInput
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Happy case",
+			args: args{
+				ctx:    ctx,
+				metric: metric,
+			},
+			wantErr: false,
+		},
+
+		{
+			name: "Sad case",
+			args: args{
+				ctx:    ctx,
+				metric: invalidMetric,
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			_, err := m.CollectMetrics(tt.args.ctx, tt.args.metric)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("UsecaseMetricsImpl.CollectMetrics() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+		})
+	}
+}
```

Comment:
Make this table-driven test.

We need to confirm actually the payload has a **HASHED** user _UUID_ and it's actually correct